src/suggestion.rs

```diff
@@ -65,9 +67,9 @@ impl Suggestion {
         }
     }
 
-    /// Get the suggestions as a slice.
-    pub fn get_suggestions(&self) -> &[String] {
-        self.suggestion.as_ref().left().unwrap()
+    /// Get the suggestions as an iterator.
+    pub fn get_suggestions(&self) -> impl Iterator<Item = &str> {
```

Comment:
Returning a `Vec<&str>` here will minimize changes in other files. Method body-
`self.suggestion.as_ref().left().unwrap().iter().map(Rank::to_string).collect()`

Reply:
I am returning an iterator here to avoid a useless allocation of a vector, as you can see the suggestions are converted into `CString` and collected in a vector again in the FFI boundary.
https://github.com/OpenBangla/riti/blob/e16a6f095d5f7dfd8e78bcd08170a1c89c57b8fe/src/ffi.rs#L139-L145

Reply:
Yes, this reduces allocations. But the maybe `vec` will increase readability and decrease diffs. But not that important though.

Reply:
I have addressed your concern partially https://github.com/OpenBangla/riti/pull/28/commits/1470381f6fc54e2c4a7210e6e77bdf38ea83c1c4